### PR TITLE
Migrate `run-make/rustdoc-with-out-dir-option` to new `rmake.rs`

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/121876.

r? @jieyouxu

### CODE REVIEW

File: tests/run-make/rustdoc-with-out-dir-option/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+use run_make_support::{htmldocck, rustdoc, tmp_dir};
+
+fn main() {
+    let out_dir = tmp_dir().join("rustdoc");
+    rustdoc().input("src/lib.rs").crate_name("foobar").crate_type("lib").output(&out_dir).run();
+    htmldocck().arg(out_dir).arg("src/lib.rs").status().unwrap().success();
```
Comment (jieyouxu) [inline tests/run-make/rustdoc-with-out-dir-option/rmake.rs:6]: I think this needs to be an assert?

Comment (GuillaumeGomez) [reply tests/run-make/rustdoc-with-out-dir-option/rmake.rs:6]: It definitely does!

**Comment by GuillaumeGomez** [comment]
I checked just in case and found two other cases where I forgot to add the `assert!`. Fixed them as well.

**Comment by GuillaumeGomez** [comment]
And that allowed me to see that the `htmldocck` function was wrong too... Well, that was a productive late evening.

**Comment by jieyouxu** [comment]
Thanks for fixing those! r=me after CI is green

**Pull request merged**